### PR TITLE
Add amlogic avc decoder to eos flush workaround list

### PR DESCRIPTION
**Pull request by unext-wendong**
A workaround for issue #2120

### CODE REVIEW

File: library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java [hunk 1]
```diff
@@ -1139,7 +1139,8 @@ private static boolean codecNeedsEosPropagationWorkaround(String name) {
    *     buffer with {@link MediaCodec#BUFFER_FLAG_END_OF_STREAM} set. False otherwise.
    */
   private static boolean codecNeedsEosFlushWorkaround(String name) {
-    return Util.SDK_INT <= 23 && "OMX.google.vorbis.decoder".equals(name);
+    return (Util.SDK_INT <= 23 && "OMX.google.vorbis.decoder".equals(name))
+        || (name != null && name.startsWith("OMX.amlogic.avc.decoder.awesome"));
```
Comment (ojw28) [inline library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java]: Unless we have evidence this affects OMX.amlogic.avc.decoder.awesome on *all* devices and API levels, this seems overly broad. Can we restrict the check to the specific device and a lower API level limit, unless we have evidence that it needs to be broader than that? For example:

return (Util.SDK_INT <= 23 && "OMX.google.vorbis.decoder".equals(name))
    || (Util.SDK_INT <= 19 && "OMX.amlogic.avc.decoder.awesome".equals(name) && Util.DEVICE.equals("xxx");

Where you need to replace xxx with whatever Util.DEVICE is equal to on the affected device. Thanks!

**Comment by ojw28** [comment]
Thanks. Happy to merge this once the comment is addressed. Note that you need to sign an appropriate CLA before we can merge the change. See [here](https://github.com/google/ExoPlayer/blob/release-v2/CONTRIBUTING.md). Please let us know once you've done this. Thanks!

File: library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java (hunk 1 above)
Comment (unext-wendong) [reply library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java]: Updated as suggested. Thanks.

**Comment by unext-wendong** [comment]
Realized I should have used my personal account. Closing this one.

Transferred to the pull request https://github.com/google/ExoPlayer/pull/2131

**Pull request closed without merging**